crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -193,38 +193,50 @@ impl BaseClient {
         ambiguity_cache: &mut AmbiguityCache,
         account_data: &AccountData,
     ) -> Result<(Option<RoomInfo>, Option<JoinedRoom>, Option<InvitedRoom>)> {
+        // If any invite_state exists, we take it to mean that we are invited to this room
+        // https://github.com/matrix-org/matrix-spec-proposals/blob/kegan/sync-v3/proposals/3575-sync.md#room-list-parameters
         if let Some(invite_state) = &room_data.invite_state {
             let room = store.get_or_create_stripped_room(room_id).await;
             let mut room_info = room.clone_info();
             room_info.mark_state_partially_synced();
 
-            let room_to_store = if let Some(r) = store.get_room(room_id) {
-                let mut stored_room_info = r.clone_info();
-                stored_room_info.mark_as_invited(); // FIXME: this might not be accurate
-                stored_room_info.mark_state_partially_synced();
-                if !room_data.required_state.is_empty() {
-                    self.handle_state(
-                        &room_data.required_state,
-                        &mut stored_room_info,
-                        changes,
-                        ambiguity_cache,
-                    )
-                    .await?;
-                }
-                Some(stored_room_info)
-            } else {
-                None
-            };
```

Comment:
I can't see this code in `main`, and I just realized this is not based off `main` but another branch of yours, is this a mistake?

Reply:
(As mentioned, it was part of a chain of PRs)